### PR TITLE
Issue #SH-597 feat: As a course mentor, I want to set rules while creating a batch (Step 1 - Issue certificate or not to the batch)

### PR DESCRIPTION
**Pull request by sudip-mukherjee**

### CODE REVIEW

File: src/app/routes/clientRoutes.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ module.exports = (app, keycloak) => {
   app.all(['/announcement', '/announcement/*', '/search', '/search/*',
   '/orgType', '/orgType/*', '/dashBoard', '/dashBoard/*',
   '/workspace', '/workspace/*', '/profile', '/profile/*', '/learn', '/learn/*', '/resources',
-  '/resources/*', '/myActivity', '/myActivity/*', '/org/*', '/manage', '/contribute','/contribute/*','/groups','/groups/*', '/my-groups','/my-groups/*'], 
+  '/resources/*', '/myActivity', '/myActivity/*', '/org/*', '/manage', '/contribute','/contribute/*','/groups','/groups/*', '/my-groups','/my-groups/*','/certs/configure/certificate'],
```
Comment (vinukumar-vs) [inline src/app/routes/clientRoutes.js:105]: It should be /certs/*

File: src/app/helpers/whitelistApis.js [hunk 2]
```diff
@@ -277,6 +278,10 @@ const API_LIST = {
       checksNeeded: ['ROLE_CHECK'],
       ROLE_CHECK: [ROLE.PUBLIC]
     },
+    '/learner/org/v2/preferences/read': {
+      checksNeeded: ['ROLE_CHECK'],
+      ROLE_CHECK: [ROLE.COURSE_MENTOR]
```
Comment (vinukumar-vs) [inline src/app/helpers/whitelistApis.js:283]: ROLE_CHECK: [ROLE.PUBLIC] This API should be publicly possible
This is for tenant preferenceApi

File: src/app/client/src/app/modules/workspace/components/update-batch/update-batch.component.ts [hunk 3]
```diff
@@ -372,7 +374,8 @@ export class UpdateBatchComponent implements OnInit, OnDestroy, AfterViewInit {
       startDate: startDate,
       endDate: endDate || null,
       createdFor: this.userService.userProfile.organisationIds,
-      mentors: _.compact(mentors)
+      mentors: _.compact(mentors),
+      cert_templates: []
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/workspace/components/update-batch/update-batch.component.ts:378]: This should not be empty always. If certificates already exist then this code is removing the cert_teamplates.

File: src/app/client/src/app/modules/shared/services/config/url.config.json [hunk 4]
```diff
@@ -302,6 +302,9 @@
     },
     "RECAPTCHA": {
       "VALIDATE": "/validate/recaptcha"
+    },
+    "TENANT_PREFERENCE" : {
+      "READ": "org/v2/preferences/read"
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/shared/services/config/url.config.json:307]: We have to onboard this API into dev-ops repo as well.

@swetha-aggidevara can you create the PR for dev-ops repo to onboard this API.

File: src/app/routes/clientRoutes.js (hunk 1 above)
Comment (sudip-mukherjee) [reply src/app/routes/clientRoutes.js:105]: if we put  /certs/* for logged in routes,  then an existing feature won't work for public users, (certificate validations)

File: src/app/client/src/app/modules/workspace/components/update-batch/update-batch.component.ts (hunk 3 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/workspace/components/update-batch/update-batch.component.ts:378]: but to remove the template anyway we need to call the template add api with empty values. i think it won't cause any issue,

File: src/app/client/src/app/modules/shared/services/config/url.config.json (hunk 4 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/shared/services/config/url.config.json:307]: this api is already been onboarded by UP team. isn't it ?

Comment (vinukumar-vs) [reply src/app/client/src/app/modules/shared/services/config/url.config.json:307]: No. We have to onboard on kong to allow the API request from portal

File: src/app/routes/clientRoutes.js (hunk 1 above)
Comment (vinukumar-vs) [reply src/app/routes/clientRoutes.js:105]: Use '/certs/configure/*'

File: src/app/client/src/app/modules/shared/services/config/url.config.json (hunk 4 above)
Comment (vinukumar-vs) [reply src/app/client/src/app/modules/shared/services/config/url.config.json:307]: Not requried

**Pull request merged**